torch/distributed/algorithms/ddp_comm_hooks/ddp_zero_hook.py

```diff
@@ -233,12 +233,12 @@ def hook_with_zero_step(
     ddp_ref = weakref.ref(ddp)
 
     # NOTE: Gloo may hang with this overlapping approach, so we require
-    # NCCL/HCCL backend for now; see https://github.com/pytorch/pytorch/issues/62300
+    # NCCL/HCCL/XCCL backend for now; see https://github.com/pytorch/pytorch/issues/62300
     pg = dist.get_backend(ddp_ref().process_group)  # type: ignore[union-attr]
-    if (pg != dist.Backend.NCCL) and (pg != "hccl"):
+    if (pg != dist.Backend.NCCL) and (pg != "hccl") and (pg != "xccl"):
```

Comment:
Shouldn't we just blacklist Gloo in the future if that is the problematic backend?